### PR TITLE
Call into all CompletionProviders in parallel to speed up completion.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/15300

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
I also tried to clean things up a lot here with simpler logic and clearer naming.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide @rchande @mattwar 

There is no change in functionality here.  It's just some refactoring and name cleanup.

File: src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs [hunk 1]
```diff
@@ -212,46 +213,51 @@ internal protected CompletionProvider GetProvider(CompletionItem item)
             var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
             var defaultItemSpan = this.GetDefaultCompletionListSpan(text, caretPosition);
 
-            options = options ?? await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);;
```
Comment (mattwar) [inline src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs]: >; [](start = 105, length = 1)

Extra semicolons mean this line of code is really important.

**Comment by mattwar** [comment]
:shipit:

File: src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs]: ;;;;;;;;;;indeed;;;;;;;;;;;;

**Pull request merged**